NRWAL/handlers/equations.py

```diff
@@ -37,18 +37,53 @@ def __init__(self, eqn, name=None, default_variables=None):
 
         self._str = None
         self._base_name = name
-        self._eqn = str(eqn)
+        self._eqn_history = [str(eqn)]
         self._preflight()
 
     def _preflight(self):
         """Run preflight checks on the equation string."""
         for substr in self.ILLEGAL:
-            if substr in str(self._eqn):
+            if substr in str(self.eqn):
                 msg = ('Will not evaluate string which contains "{}": {}'
-                       .format(substr, self._eqn))
+                       .format(substr, self.eqn))
                 logger.error(msg)
                 raise ValueError(msg)
 
+        self.verify_no_self_reference()
+
+    @property
+    def eqn(self):
+        """str: Full equation string
+
+        The setter method verifies that the new equation
+        does not contain any self references.
+        """
+        return self._eqn_history[-1]
+
+    @eqn.setter
+    def eqn(self, value):
```

Comment:
I'm a bit wary of this. Have you experienced any cases where you want to replace the underlying equation? Why wouldn't you just create a new Equation object? I'm afraid this could open us up to undesirable use cases.

Reply:
Yes, the EquationClass object was previously setting the protected attribute `_eqn` on line 753 of `groups.py`: https://github.com/NREL/NRWAL/blob/f7919b40b2cc0cdf5abd579ffe2e3788687cd7c2/NRWAL/handlers/groups.py#L753
It looks like the particular use case is to perform variable substitutions. Would you prefer that I add an `Equation` method to handle variable substitution? That would remove the need to set `eqn` directly

Reply:
Yeah i think that's fine actually. IMO we're allowed to touch and manipulate protected attributes, we just don't want an external user of NRWAL to do that. 

I just feel like if you're replacing the underlying eqn string you may as well init a new Equation object to make sure the full init runs. Maybe thats the solution in groups? Just initialize a new Equation object?

Reply:
Ok, let me try that out!

Reply:
Thinking about this more, it might make sense if you added a classmethod that was something like `replace_equation()` that would handle any of the optional kwargs to `Equation.__init__()`

Reply:
classmethod would be nice because it would return a new instance of cls (Equation) and provide an obvious place to maintain any kwarg passing

Reply:
That's a good call, the code was starting to look a bit ugly with the kwarg passing. I will do that now!

Reply:
I still dont love that you need to pass in the old equation object haha. I'm starting to think you may have been right all along. Or what if we do this:
```
def replace_eqn(self, new_eqn):
    return self.__class__(new_eqn, self.kwargs)
```
A bit hacky but the interface is really clean!

Reply:
On it!